server/task/task-distributed/src/main/scala/org/apache/james/task/eventsourcing/cassandra/CassandraTaskExecutionDetailsProjectionDAO.scala

```diff
@@ -76,17 +89,26 @@ class CassandraTaskExecutionDetailsProjectionDAO(session: Session, typesProvider
     .executeRows(listStatement.bind())
     .map(readRow)
 
-  private def readRow(row: Row): TaskExecutionDetails = new TaskExecutionDetails(
-    taskId = TaskId.fromUUID(row.getUUID(TASK_ID)),
-    `type` = TaskType.of(row.getString(TYPE)),
-    status = TaskManager.Status.fromString(row.getString(STATUS)),
-    submittedDate = CassandraZonedDateTimeModule.fromUDT(row.getUDTValue(SUBMITTED_DATE)),
-    submittedNode = Hostname(row.getString(SUBMITTED_NODE)),
-    startedDate = CassandraZonedDateTimeModule.fromUDTOptional(row.getUDTValue(STARTED_DATE)),
-    ranNode = Optional.ofNullable(row.getString(RAN_NODE)).map(Hostname(_)),
-    completedDate = CassandraZonedDateTimeModule.fromUDTOptional(row.getUDTValue(COMPLETED_DATE)),
-    canceledDate = CassandraZonedDateTimeModule.fromUDTOptional(row.getUDTValue(CANCELED_DATE)),
-    cancelRequestedNode = Optional.ofNullable(row.getString(CANCEL_REQUESTED_NODE)).map(Hostname(_)),
-    failedDate = CassandraZonedDateTimeModule.fromUDTOptional(row.getUDTValue(FAILED_DATE)),
-    additionalInformation = Optional.empty)
+  private def readRow(row: Row): TaskExecutionDetails = {
+    val taskType = TaskType.of(row.getString(TYPE))
+    new TaskExecutionDetails(
+      taskId = TaskId.fromUUID(row.getUUID(TASK_ID)),
+      `type` = TaskType.of(row.getString(TYPE)),
+      status = TaskManager.Status.fromString(row.getString(STATUS)),
+      submittedDate = CassandraZonedDateTimeModule.fromUDT(row.getUDTValue(SUBMITTED_DATE)),
+      submittedNode = Hostname(row.getString(SUBMITTED_NODE)),
+      startedDate = CassandraZonedDateTimeModule.fromUDTOptional(row.getUDTValue(STARTED_DATE)),
+      ranNode = Optional.ofNullable(row.getString(RAN_NODE)).map(Hostname(_)),
+      completedDate = CassandraZonedDateTimeModule.fromUDTOptional(row.getUDTValue(COMPLETED_DATE)),
+      canceledDate = CassandraZonedDateTimeModule.fromUDTOptional(row.getUDTValue(CANCELED_DATE)),
+      cancelRequestedNode = Optional.ofNullable(row.getString(CANCEL_REQUESTED_NODE)).map(Hostname(_)),
+      failedDate = CassandraZonedDateTimeModule.fromUDTOptional(row.getUDTValue(FAILED_DATE)),
+      additionalInformation = () => deserializeAdditionalInformations(taskType, row),
+    )
+  }
+
+  private def deserializeAdditionalInformations(taskType: TaskType, row: Row): Optional[TaskExecutionDetails.AdditionalInformation] = {
```

Comment:
I don't think that there is a style to choose for the return type of private method.
For me : 
- for public method the return type is mandatory
- for private method it's optional, and we could add when : 
    - it helps understand the method
    - it helps the compiler or remove ambiguity between types

Reply:
I can see the value of not having a rule but let me know what you think about this case: serializeAdditionalInformations doesn't declare a type and deserializeAdditionalInformations does. Does it make sense?

Reply:
we declare a type in the `deserializeAdditionalInformations` method in order to help the compiler because if we don't do it it infers the return type to `Optional[Nothing]` instead of `Optional[TaskExecutionDetails.AdditionalInformation]`

Reply:
> my suggestion is to put types everywhere it improves readability. So I guess everywhere :)

Nothing to read is often more readable than something to read so I think the opposite